### PR TITLE
Deprecate HttplugFactory and make dependency on php-http/message-factory optional

### PR DESCRIPTION
**Pull request by murat11**
Follows https://github.com/php-http/message-factory/pull/42

### CODE REVIEW

File: src/Factory/HttplugFactory.php [hunk 1]
```diff
@@ -17,7 +20,7 @@
  *
  * @final This class should never be extended. See https://github.com/Nyholm/psr7/blob/master/doc/final.md
  */
-class HttplugFactory implements MessageFactory, StreamFactory, UriFactory
+class HttplugFactory implements RequestFactoryInterface, ResponseFactoryInterface, StreamFactoryInterface, UriFactoryInterface
 {
     public function createRequest($method, $uri, array $headers = [], $body = null, $protocolVersion = '1.1'): RequestInterface
```
[the remark this answers is not in the document]
Comment (nicolas-grekas) [reply src/Factory/HttplugFactory.php:25]: no need to care since we can deprecate the whole class instead

[the remark this answers is not in the document]
Comment (murat11) [reply src/Factory/HttplugFactory.php]: okay

**Review by nicolas-grekas — APPROVED** [review]
I updated the patch to account for a few more things, especially making the dependency on php-http/message-factory optional.

This PR is motivated by https://github.com/php-http/message-factory/pull/42

**Comment by nicolas-grekas** [comment]
Thank you @murat11

**Pull request merged**